kubernetes/types.go

```diff
@@ -57,40 +57,25 @@ const (
 	// K8s Networking
 
 	K8sGateways    = "k8sgateways"
-	K8sGatewayType = "K8sGateway"
-	// K8sActualGatewayType There is a naming conflict between Istio and K8s Gateways, keeping here an actual type to show in YAML editor
-	K8sActualGatewayType = "Gateway"
-	K8sActualGateways    = "gateways"
+	K8sGatewayType = "Gateway"
 
-	K8sGatewayClasses       = "k8sgatewayclasses"
-	K8sGatewayClassType     = "GatewayClass"
-	K8sActualGatewayClasses = "gatewayclasses"
+	K8sGatewayClasses   = "k8sgatewayclasses"
```

Comment:
It looks like we're still keeping some of the `k8s...` types. I was expecting this change to remove those and replace them with the real GVK e.g. `gateway.networking.k8s.io/v1.Gateway`. I haven't looked at it very closely but it doesn't seem like we need to define any of the types/mappings here in `kubernete/types.go` since those are already defined in the api packages themselves.

Are we going to show the full GVK on the frontend as well? For example this:
![Screenshot from 2024-08-16 09-45-20](https://github.com/user-attachments/assets/45176c5a-2639-4e26-b95d-ceef68bfb887)

would become something like `Gateway (gateway.networking.k8s.io/v1)` or `gateway.networking.k8s.io/v1.Gateway` etc. since that is the real type.

Reply:
yeah we still have them, because of istio config details page, when the url is '/namespaces/namespace/configtype/configname'

Reply:
If we want to completely remove `K8sGateways    = "k8sgateways"` and the rest, then we need to rework `handlers.IstioConfigDetails /api/namespaces/{namespace}/istio/{object_type}/{object}` and similar REST endpoints for Istoi config.

Reply:
Yes the endpoints would need to be updated in order to fully remove `k8sgateways`. If we wanted to just display the real type to the user we could maintain the mapping on the backend along with the routes. The frontend should always have the full object so it can use `.apiVersion` and `.kind`. If we're going to close #7457 with this change can we create a separate issue to track displaying the real type on the frontend and possibly reworking the routes?